Sources/ClientGtk/ServerListView.swift

```diff
@@ -41,61 +46,135 @@ struct ServerListView: View {
     } detail: {
       VStack {
         switch state.detailState {
-          case .addition:
-            additionView
-          case .server(let server):
-            Text(server.name)
-            if let port = server.port {
-              Text("\(server.host):\(port)")
-            } else {
-              Text(server.host)
-            }
-
-            Button("Connect") {
-              completionHandler(server)
-            }
-          case .error(let message):
+          case .server(let index):
+            serverView(index)
+          case .adding:
+            addingView()
+          case .editing(let index):
+            editingView(index)
+          case .error(let message, let returnState):
             Text("Error: \(message)")
-            Button ("Back") {
-              state.detailState = .addition
+            if let returnState {
+              Button ("Back") {
+                state.detailState = returnState
+              }
             }
         }
       }
       .padding(.leading, 10)
     }
   }
 
-  var additionView: some View {
-    VStack {
+  func serverView(_ index: Int) -> some View {
+    let server = state.servers[index]
+
+    return VStack {
+      Text(server.name)
+      if let port = server.port {
+        Text("\(server.host):\(port)")
+      } else {
+        Text(server.host)
+      }
+
+      Button("Connect") {
+        completionHandler(server)
+      }
+
+      Button("Edit") {
+        state.name = server.name
+        if let port = server.port {
```

Comment:
You can use `state.address = server.description` since it happens that I've already implemented this logic